### ISSUE TITLE
RK-S98 Not Supported by Rangoli

### ISSUE THREAD

**Issue by nogames5613** [rnayabed/rangoli#79]
Hey!

Hope all is well!

I recently purchased the RK-S98 form Royal Kludge and the software on their website seems to not support this model or something. I found your software from a reddit page and was advised that Rangoli was better or more supportive. But alas it seems that this model isn't listed in the supported items yet. I'm unsure if this is the case for all S98 models or not but I'm guessing because it's a new model that I'm having these difficulties? I've seen videos from YT and Reddit where others were able to use the software with their keyboard but those were made Months or even Years ago.

I'm at my wits end with this and I'm hoping that you have an update or a way for me to get the keyboard to work with your software so I can make some updates to the keyboard like the time, colors, gifs, etc.

Thanks for all your time, looking forward to hearing from you!

**Comment by DerPandaa1** [comment]
Duplicate of #68. A workaround could be copying the 143.json and rename it to 174.json and change the pid to 174. Then copy the 143.png and rename it to 174.png. This supports only RGB and Keymapping. My S98 will be delivered next week. Then i will work on support for gifs and use the right image.

**Comment by nogames5613** [comment]
I'm not seeing te 143.json file either. The highest I see is 103.json in C:\Program Files\Rangoli\bin\keyboards\258a\configs. I could be looking in the wrong place maybe, or downloaded an older version of Rangoli?

**Comment by nogames5613** [comment]
I do see the 143.json and png files in the commits but not after downloading the .exe

**Comment by nogames5613** [comment]
So I'm a dummy lol but I was able to get the software to recognize the keyboard...BUT now I'm getting a 'Failed to send feature report to keyboard. Make sure you have sufficient permissions.' error

![Screenshot 2024-02-14 190942](https://github.com/rnayabed/rangoli/assets/159677002/8c72874e-8d40-40ce-9974-5cb7a0867f2c)

**Comment by DerPandaa1** [comment]
Maybe you can document how you got it working. It would help me to make the changes faster. But im still waiting for delivery. Maybe Germany isnt the best land to deliver.
Ideas to fix the issue is run it as administrator. Otherwise i have no idea and we have to wait for my delivery

**Comment by nogames5613** [comment]
So I did the work around you suggested where I copied the 143.json file to 174.json and added it to the config file. I found a png image for the keyboard and just put that in the images directory for it load the correct image for the S98. I will say, and this was my slow moment, that you MUST have the keyboard set to 'wired mode' in order for the PC/software to recognize the device. I had mine plugged in but set to Bluetooth mode so the computer and software didn't recognize/find my keyboard. Once I switched modes, I was able to use the software.

But alas, when I was in the software, I wasn't able to use any of the features due to the error messages above. I'm thinking it could be due to the .json file copied from 143 but unsure. I did change the PID for the new jsaon file but still no luck. 

The factory software is beyond outdated so will definitely like to work with you to get this work in order for other to use your amazing UI and software!

**Comment by DerPandaa1** [comment]
so after 2 Weeks. I finally got my keyboard. I want to say I'm not a maintainer of this software, I only want my Keyboard to work. I tried to get it running but it doesn't recognize my keyboard. Maybe i need your help to get it working to work on the GIFs. Your Problem seems like it doesn't have sudo/admin.

**Comment by nogames5613** [comment]
If you were able to get yours to work let me know! I was talking with one of the devs I think and still waiting on a response as far as their progress.

**Comment by Reaper-EXE** [comment]
Did somebody got this working on linux?

**Comment by melpst** [comment]
> so after 2 Weeks. I finally got my keyboard. I want to say I'm not a maintainer of this software, I only want my Keyboard to work. I tried to get it running but it doesn't recognize my keyboard. Maybe i need your help to get it working to work on the GIFs. Your Problem seems like it doesn't have sudo/admin.

@DerPandaa1 for me, as mac user, i checked the pid of RK-S98 from hidapi directly and got '01af' instead of '0174', so i changed to the first one instead
after that rangoli can detect my keyboard but then i get the same error as @nogames5613 

> So I'm a dummy lol but I was able to get the software to recognize the keyboard...BUT now I'm getting a 'Failed to send feature report to keyboard. Make sure you have sufficient permissions.' error
> 
> ![Screenshot 2024-02-14 190942](https://private-user-images.githubusercontent.com/159677002/304919705-8c72874e-8d40-40ce-9974-5cb7a0867f2c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM1Nzc5MzUsIm5iZiI6MTczMzU3NzYzNSwicGF0aCI6Ii8xNTk2NzcwMDIvMzA0OTE5NzA1LThjNzI4NzRlLThkNDAtNDBjZS05OTc0LTVjYjdhMDg2N2YyYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIwN1QxMzIwMzVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yN2YxYTIxMTM4MjViNzVkZGM1OTFmNDdiMThhMDNmMmFmZTBlZGRlNzZjYzU3NmIwMDM5ZDY5NmI2MjIxYjRmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.0zY-CPpqvGrvBqnLHHxmPri5pUhx8F0sIkho1SbMacw)

i even tried to open rangoli with sudo but it still doesn't work :( so i think it's more like a coding issue. maybe need to check how hidapi send settings to RK-S98...
